src/sql/src/plan/query.rs

```diff
@@ -905,6 +905,52 @@ pub fn plan_secret_as(
     Ok(expr)
 }
 
+/// Plans an expression in the VALIDATE USING position of a `CREATE SOURCE ... FROM WEBHOOK`.
+pub fn plan_webhook_validate_using(
+    scx: &StatementContext,
+    mut expr: Expr<Aug>,
+) -> Result<MirScalarExpr, PlanError> {
+    let qcx = QueryContext::root(scx, QueryLifetime::OneShot(scx.pcx()?));
```

Comment:
Maybe this should be `QueryLifetime::Static` because it's not computed at a single point in time. `VALIDATE USING current_timestamp()` for example shouldn't work. I think this should fail if it includes unmaterializable functions.